### PR TITLE
Add support for localised Snap manifests

### PR DESCRIPTION
**Pull request by Mrtenz**
## **Description**

This adds support for localised Snap manifests. Instead of getting the Snap name from the subject metadata controller, I've added a new selector which returns the localised manifest.

Related to MetaMask/snaps#1508.

## **Manual testing steps**

1. Install `npm:@metamask/localization-example-snap` from test-snaps.
2. Make sure the proper name and description is shown everywhere.

## **Screenshots/Recordings**

image

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by Mrtenz** [comment]
@metamaskbot update-policies

File: ui/helpers/utils/util.js [hunk 1]
```diff
@@ -613,14 +610,6 @@ export function getSnapDerivationPathName(path, curve) {
 export const removeSnapIdPrefix = (snapId) =>
   snapId?.replace(getSnapPrefix(snapId), '');
 
-export const getSnapName = (snapId, subjectMetadata) => {
-  if (SNAPS_METADATA[snapId]?.name) {
```
Comment (FrederikBolding) [inline ui/helpers/utils/util.js:617]: We can probably delete `SNAPS_METADATA` then right?

File: ui/pages/confirmation/confirmation.js [hunk 2]
```diff
@@ -21,22 +21,19 @@ import { DEFAULT_ROUTE } from '../../helpers/constants/routes';
...
 import {
-  ///: BEGIN:ONLY_INCLUDE_IN(snaps)
-  getTargetSubjectMetadata,
-  ///: END:ONLY_INCLUDE_IN
   getUnapprovedTemplatedConfirmations,
   getUnapprovedTxCount,
   getApprovalFlows,
   getTotalUnapprovedCount,
   useSafeChainsListValidationSelector,
+  getSnapMetadata,
```
Comment (FrederikBolding) [inline ui/pages/confirmation/confirmation.js:29]: Missing fencing

File: ui/hooks/useTransactionInsights.js [hunk 3]
```diff
@@ -29,8 +28,22 @@ const useTransactionInsights = ({ txData }) => {
   const { txParams, chainId, origin } = txData;
   const caip2ChainId = `eip155:${stripHexPrefix(chainId)}`;
   const insightSnaps = useSelector(getInsightSnaps);
-  const subjectMetadata = useSelector(getSubjectMetadataDeepEqual);
   const insightSnapIds = useSelector(getInsightSnapIds);
+  const state = useSelector((value) => value);
```
Comment (FrederikBolding) [inline ui/hooks/useTransactionInsights.js:32]: I wonder if this will cause re-rendering problems, if so that is a merge blocker

**Review by FrederikBolding — CHANGES_REQUESTED** [review]

File: ui/helpers/utils/util.js (hunk 1 above)
Comment (Mrtenz) [reply ui/helpers/utils/util.js:617]: I don't think we use it (for anything other than some example Snaps), so might as well remove it.

File: ui/hooks/useTransactionInsights.js (hunk 3 above)
Comment (Mrtenz) [reply ui/hooks/useTransactionInsights.js:32]: Potentially yeah. Do you have better ideas?

Comment (FrederikBolding) [reply ui/hooks/useTransactionInsights.js:32]: Potentially we could create a selector that gets the manifests for all transaction insight snaps and use that instead of a function?

File: ui/helpers/utils/util.js (hunk 1 above)
Comment (FrederikBolding) [reply ui/helpers/utils/util.js:617]: Exactly!

File: ui/hooks/useTransactionInsights.js (hunk 3 above)
Comment (hmalik88) [reply ui/hooks/useTransactionInsights.js]: Done.

File: app/scripts/metamask-controller.js [hunk 4]
```diff
@@ -2388,6 +2389,30 @@ export default class MetamaskController extends EventEmitter {
     );
 
     ///: BEGIN:ONLY_INCLUDE_IF(snaps)
+    // Record Snap metadata whenever a Snap is added to state.
+    this.controllerMessenger.subscribe(
```
Comment (FrederikBolding) [inline app/scripts/metamask-controller.js:2393]: This should be removed and upstreamed to: https://github.com/MetaMask/snaps/blob/main/packages/snaps-controllers/src/snaps/SnapController.ts#L2574-L2580

File: ui/helpers/utils/permission.js [hunk 5]
```diff
@@ -245,88 +244,68 @@ export const PERMISSION_DESCRIPTIONS = deepFreeze({
...
-              key={snapId}
-            >
-              {friendlyName}
-            </Text>,
-          ]),
-          description: t('permission_accessSnapDescription', [friendlyName]),
-        };
-      }
-
+      const friendlyName = snapName;
```
Comment (FrederikBolding) [inline ui/helpers/utils/permission.js:280]: This is flawed. The name here should be the name of the requested snap as part of the permission, not the installing snap. Also we may not have a name for this snap, so a bunch of code needs to be reverted here

File: ui/selectors/selectors.js [hunk 6]
```diff
@@ -1089,6 +1089,50 @@ export const getSnap = createDeepEqualSelector(
...
+
+/**
+ * Get a selector that returns the snap metadata (name and description) for a
+ * given `snapId`.
+ *
+ * @param {object} state - The Redux state object.
+ * @param {string} snapId - The snap ID to get the metadata for.
+ * @returns {object} An object containing the snap name and description.
+ */
+export const getSnapMetadata = createDeepEqualSelector(
```
Comment (FrederikBolding) [inline ui/selectors/selectors.js:1123]: Maybe we need a selector that contains all the metadata for all snaps? That would be easily passed down through the permission copy logic. We can keep one for accessing just the metadata of one snap though

File: ui/selectors/selectors.js [hunk 7]
```diff
@@ -1089,6 +1089,50 @@ export const getSnap = createDeepEqualSelector(
...
+  (state, snapId) => getSnapManifest(state, snapId),
+  (_, snapId) => snapId,
+  (manifest, snapId) => {
+    return {
+      // The snap manifest may not be available if the Snap is not installed, so
+      // we use the snap ID as the name in that case.
+      name: manifest?.proposedName ?? stripSnapPrefix(snapId),
```
Comment (FrederikBolding) [inline ui/selectors/selectors.js:1130]: IIRC `snapId` is occasionally null when we get a too quick re-render. We may want to guard against that. See previous `getSnapName` function

File: lavamoat/browserify/main/policy.json [hunk 8]
```diff
@@ -709,14 +709,22 @@
     },
```
Comment (FrederikBolding) [inline lavamoat/browserify/main/policy.json:709]: I don't think there should be policy changes in this PR 🤔

**Review by FrederikBolding — CHANGES_REQUESTED** [review]

File: app/scripts/metamask-controller.js (hunk 4 above)
Comment (hmalik88) [reply app/scripts/metamask-controller.js:2393]: Done https://github.com/MetaMask/snaps/pull/2157

File: ui/selectors/selectors.js (hunk 7 above)
Comment (hmalik88) [reply ui/selectors/selectors.js:1172]: I don't see a guard against it in the previous function?

File: ui/selectors/selectors.js (hunk 6 above)
Comment (hmalik88) [reply ui/selectors/selectors.js:1165]: ok, will add.

File: ui/selectors/selectors.js (hunk 7 above)
Comment (FrederikBolding) [reply ui/selectors/selectors.js:1192]: That's what this is: https://github.com/MetaMask/metamask-extension/blob/201a48fd76e89c68ea9a559d10dc58fa14d7da61/ui/helpers/utils/util.js#L577-L580

File: ui/components/app/snaps/snap-ui-renderer/snap-ui-renderer.js [hunk 9]
```diff
@@ -38,12 +33,11 @@ const SnapUIRendererComponent = ({
...
   const t = useI18nContext();
-  const targetSubjectMetadata = useSelector((state) =>
-    getMemoizedTargetSubjectMetadata(state, snapId),
+  const { name: snapName } = useSelector((state) =>
+    getSnapMetadata(state, snapId),
   );
 
-  const snapName = getSnapName(snapId, targetSubjectMetadata);
-
+  if (isLoading) {
```
Comment (FrederikBolding) [inline ui/components/app/snaps/snap-ui-renderer/snap-ui-renderer.js:40]:
```suggestion
```
Not sure what this addition is?

File: app/_locales/de/messages.json [hunk 10]
```diff
@@ -2836,10 +2836,6 @@
     "message": "Greifen Sie auf das Internet zu.",
     "description": "The description of the `endowment:network-access` permission."
   },
-  "permission_accessSnap": {
```
Comment (FrederikBolding) [inline app/_locales/de/messages.json:2839]: These changes to messages should be reverted

File: lavamoat/browserify/main/policy.json [hunk 11]
```diff
@@ -817,6 +824,32 @@
         "webpack>events": true
       }
     },
+<<<<<<< HEAD
+    "@metamask/assets-controllers>@metamask/abi-utils": {
```
Comment (FrederikBolding) [inline lavamoat/browserify/main/policy.json:828]: Please revert the policy changes and regen using the bot

File: ui/components/app/snaps/snap-metadata-modal/snap-metadata-modal.js [hunk 12]
```diff
@@ -49,6 +50,10 @@ export const SnapMetadataModal = ({ snapId, isOpen, onClose }) => {
     getTargetSubjectMetadata(state, snapId),
   );
 
+  const { name: snapName } = useSelector((state) =>
```
Comment (FrederikBolding) [inline ui/components/app/snaps/snap-metadata-modal/snap-metadata-modal.js:53]: We should use the translated description in this component

File: ui/ducks/domains.js [hunk 13]
```diff
@@ -276,6 +277,9 @@ export function lookupDomainName(domainName) {
         }
       }
 
+      const { snapId } = fetchResolutions[0];
```
Comment (FrederikBolding) [inline ui/ducks/domains.js:280]:
```suggestion
      const snapId = fetchedResolutions?.[0].snapId;
```

File: ui/components/app/snaps/snap-permissions-list/snap-permissions-list.js [hunk 14]
```diff
@@ -7,19 +7,21 @@ import { Box } from '../../../component-library';
...
   showOptions,
 }) {
   const t = useI18nContext();
 
   return (
     <Box paddingTop={2} paddingBottom={2} className="snap-permissions-list">
-      {getWeightedPermissions(t, permissions, targetSubjectMetadata).map(
-        (permission, index) => {
+      {getWeightedPermissions({ t, permissions, snapName })
+        .map()
```
Comment (FrederikBolding) [inline ui/components/app/snaps/snap-permissions-list/snap-permissions-list.js:19]:
```suggestion
```

File: ui/helpers/utils/permission.js [hunk 15]
```diff
@@ -649,16 +617,28 @@ export const getPermissionDescription = ({
...
- * @param {object} targetSubjectMetadata - The subject metadata.
+ * @param {WeightedPermissionDescriptionParamsObject} parms - The weighted permissions params object.
  * @returns {PermissionLabelObject[]}
  */
-export function getWeightedPermissions(t, permissions, targetSubjectMetadata) {
+export function getWeightedPermissions({
+  t,
+  permissions,
+  getSnapName,
+  snapName,
```
Comment (FrederikBolding) [inline ui/helpers/utils/permission.js:640]: Should this be named more generically? E.g. `subjectName`?

File: ui/selectors/selectors.js [hunk 16]
```diff
@@ -1326,6 +1390,29 @@ export const getInsightSnaps = createDeepEqualSelector(
   },
 );
 
+/**
+ * Get a selector that returns insight snaps metadata (name and description).
+ *
+ * @param {object} state - The Redux state object.
+ * @returns {object} An object with snapId keys and values of snap names and descriptions.
+ */
+export const getInsightSnapsMetadata = createDeepEqualSelector(
```
Comment (FrederikBolding) [inline ui/selectors/selectors.js:1399]: Why do we need this and `getSnapsMetadata`? 🤔

Comment (hmalik88) [reply ui/selectors/selectors.js:1399]: You're right lol, not needed. Removed!

File: ui/pages/snaps/snap-view/snap-settings.js [hunk 17]
```diff
@@ -37,13 +37,12 @@ import {
   getSnaps,
   getSubjectsWithSnapPermission,
   getPermissions,
-  getTargetSubjectMetadata,
   getSnapLatestVersion,
   ///: BEGIN:ONLY_INCLUDE_IF(keyring-snaps)
   getMemoizedMetaMaskIdentities,
+  getSnapMetadata,
```
Comment (FrederikBolding) [inline ui/pages/snaps/snap-view/snap-settings.js:43]: This needs to be moved out of the fencing

File: ui/ducks/domains.js [hunk 18]
```diff
@@ -276,6 +277,9 @@ export function lookupDomainName(domainName) {
         }
       }
 
+      const snapId = fetchResolutions?.[0].snapId;
```
Comment (FrederikBolding) [inline ui/ducks/domains.js:280]:
```suggestion
      const snapId = fetchedResolutions?.[0].snapId;
```

File: ui/components/app/snaps/snap-permissions-list/snap-permissions-list.js [hunk 19]
```diff
@@ -7,19 +7,20 @@ import { Box } from '../../../component-library';
 
 export default function SnapPermissionsList({
   snapId,
+  snapName,
   permissions,
-  targetSubjectMetadata,
   showOptions,
 }) {
   const t = useI18nContext();
```
Comment (FrederikBolding) [inline ui/components/app/snaps/snap-permissions-list/snap-permissions-list.js:14]: Wondering if it would be easier for the SnapPermissionList to get the Snap name from the ID?

File: ui/components/app/snaps/snap-permissions-list/snap-permissions-list.js [hunk 20]
```diff
@@ -7,19 +7,20 @@ import { Box } from '../../../component-library';
...
 
   return (
     <Box paddingTop={2} paddingBottom={2} className="snap-permissions-list">
-      {getWeightedPermissions(t, permissions, targetSubjectMetadata).map(
+      {getWeightedPermissions({ t, permissions, subjectName: snapName }).map(
         (permission, index) => {
           return (
             <PermissionCell
               snapId={snapId}
+              snapName={snapName}
```
Comment (FrederikBolding) [inline ui/components/app/snaps/snap-permissions-list/snap-permissions-list.js:23]: This prop doesn't exist

File: ui/components/app/snaps/update-snap-permission-list/update-snap-permission-list.js [hunk 21]
```diff
@@ -13,40 +15,48 @@ export default function UpdateSnapPermissionList({
...
-            dateApproved={permission?.permissionValue?.date}
-            key={`${permission.permissionName}-${index}`}
-            revoked
-          />
-        ),
-      )}
-      {getWeightedPermissions(
+      {getWeightedPermissions({
+        t,
+        permissions: newPermissions,
```
Comment (FrederikBolding) [inline ui/components/app/snaps/update-snap-permission-list/update-snap-permission-list.js:26]: We need to pass `getSubjectName` to all instances of `getWeightedPermissions`

File: app/scripts/metamask-controller.js [hunk 22]
```diff
@@ -986,6 +992,30 @@ export default class MetamaskController extends EventEmitter {
       await this.accountsController.updateAccounts();
     };
 
+    const getSnapName = (id) => {
+      if (!id) {
+        return null;
+      }
+
+      const { currentLocale, snaps } = this.store.getState();
```
Comment (FrederikBolding) [inline app/scripts/metamask-controller.js:1000]:
```suggestion
      const currentLocale = this.getLocale()
      const { snaps } = this.snapController.state;
```

File: app/scripts/metamask-controller.js [hunk 23]
```diff
@@ -986,6 +992,30 @@ export default class MetamaskController extends EventEmitter {
...
+      if (snap.localizationFiles) {
+        const localizedManifest = getLocalizedSnapManifest(
+          snap.manifest,
+          currentLocale,
+          snap.localizationFiles,
+        );
+        return localizedManifest.name;
+      }
+
+      return snap.manifest.name;
```
Comment (FrederikBolding) [inline app/scripts/metamask-controller.js:1016]:
```suggestion
      return snap.manifest.proposedName;
```

File: app/scripts/metamask-controller.js [hunk 24]
```diff
@@ -986,6 +992,30 @@ export default class MetamaskController extends EventEmitter {
...
+        return stripSnapPrefix(id);
+      }
+
+      if (snap.localizationFiles) {
+        const localizedManifest = getLocalizedSnapManifest(
+          snap.manifest,
+          currentLocale,
+          snap.localizationFiles,
+        );
+        return localizedManifest.name;
```
Comment (FrederikBolding) [inline app/scripts/metamask-controller.js:1013]:
```suggestion
        return localizedManifest.proposedName;
```

File: ui/pages/confirmations/confirmation/confirmation.js [hunk 25]
```diff
@@ -241,9 +241,14 @@ export default function ConfirmationPage({
   const [submitAlerts, setSubmitAlerts] = useState([]);
 
   ///: BEGIN:ONLY_INCLUDE_IF(snaps)
-  const targetSubjectMetadata = useSelector((state) =>
-    getTargetSubjectMetadata(state, pendingConfirmation?.origin),
-  );
+  const snapsMetadata = useSelector((state) => {
+    if (getSnaps(state)) {
```
[the remark this answers is not in the document]
Comment (hmalik88) [reply ui/pages/confirmations/confirmation/confirmation.js:245]: Doing some wonky stuff here to get around tests. I'm going to revert and actually add in some dummy data 😄

File: ui/components/app/snaps/snap-permissions-list/snap-permissions-list.js (hunk 19 above)
Comment (hmalik88) [reply ui/components/app/snaps/snap-permissions-list/snap-permissions-list.js:14]: I kept it like this because the settings page uses the `snapName` quite a bit and I didn't see the sense in recalculating it.

File: ui/components/app/snaps/update-snap-permission-list/update-snap-permission-list.js (hunk 21 above)
Comment (hmalik88) [reply ui/components/app/snaps/update-snap-permission-list/update-snap-permission-list.js:26]: Why? we're providing the `subjectName` already. The `getSubjectName` function is only really used by the `wallet_snap` permission.

Comment (FrederikBolding) [reply ui/components/app/snaps/update-snap-permission-list/update-snap-permission-list.js:26]: Yeah, but the `wallet_snap` permission can be used everywhere right?

Comment (hmalik88) [reply ui/components/app/snaps/update-snap-permission-list/update-snap-permission-list.js:26]: oh right, snap dependencies. I'll fix.

**Review by FrederikBolding — DISMISSED** [review]

**Review by FrederikBolding — DISMISSED** [review]

File: ui/components/app/permissions-connect-permission-list/permissions-connect-permission-list.js [hunk 26]
```diff
@@ -25,22 +28,25 @@ function getDescriptionNode(permission, index) {
...
-  permissions,
-  targetSubjectMetadata,
-}) {
+export default function PermissionsConnectPermissionList({ permissions }) {
   const t = useI18nContext();
+  const snapsMetadata = useSelector(getSnapsMetadata);
+
+  const getSnapName = (id) => {
+    return snapsMetadata[id]?.name ?? stripSnapPrefix(id);
+  };
```
Comment (GuillaumeRx) [inline ui/components/app/permissions-connect-permission-list/permissions-connect-permission-list.js:37]: Can we extract this and add it to a utils file ? This function is redefined in multiple places

File: ui/components/app/snaps/snap-authorship-header/snap-authorship-header.js [hunk 27]
```diff
@@ -33,12 +32,10 @@ const SnapAuthorshipHeader = ({
   // like it is done with snap install
   const packageName = snapId && stripSnapPrefix(snapId);
 
-  const subjectMetadata = useSelector((state) =>
-    getTargetSubjectMetadata(state, snapId),
+  const { name: snapName } = useSelector((state) =>
+    getSnapMetadata(state, snapId),
   );
```
Comment (GuillaumeRx) [inline ui/components/app/snaps/snap-authorship-header/snap-authorship-header.js:38]: Can this be uniform ? Sometimes you renamed it to `snapName` and sometimes `friendlyName`

File: ui/hooks/useTransactionInsights.js [hunk 28]
```diff
@@ -98,7 +110,7 @@ const useTransactionInsights = ({ txData }) => {
     insightComponent = (
       <Tab
         className="confirm-page-container-content__tab"
-        name={getSnapName(selectedSnap?.id, subjectMetadata[selectedSnap?.id])}
+        name={getSnapName(selectedSnap?.id)}
```
Comment (GuillaumeRx) [inline ui/hooks/useTransactionInsights.js:113]: Can this be undefined ? Before this would never be undefined 🤔

File: ui/pages/confirmations/confirmation/confirmation.js [hunk 29]
```diff
@@ -241,9 +240,9 @@ export default function ConfirmationPage({
   const [submitAlerts, setSubmitAlerts] = useState([]);
 
   ///: BEGIN:ONLY_INCLUDE_IF(snaps)
-  const targetSubjectMetadata = useSelector((state) =>
-    getTargetSubjectMetadata(state, pendingConfirmation?.origin),
-  );
+  const snapsMetadata = useSelector(getSnapsMetadata);
+
+  const name = snapsMetadata[pendingConfirmation?.origin]?.name;
```
Comment (GuillaumeRx) [inline ui/pages/confirmations/confirmation/confirmation.js:245]: Can't we do this ?

```suggestion
  const { name } = useSelector((state) => getSnapMetadata(state, pendingConfirmation?.origin));
```

File: ui/selectors/selectors.js [hunk 30]
```diff
@@ -1307,6 +1307,70 @@ export const getSnap = createDeepEqualSelector(
...
+ * @returns {object} An object containing the snap name and description.
+ */
+export const getSnapMetadata = createDeepEqualSelector(
+  (state, snapId) => getSnapManifest(state, snapId),
+  (_, snapId) => snapId,
+  (manifest, snapId) => {
+    return {
+      // The snap manifest may not be available if the Snap is not installed, so
+      // we use the snap ID as the name in that case.
+      name: manifest?.proposedName ?? (snapId ? stripSnapPrefix(snapId) : null),
```
Comment (GuillaumeRx) [inline ui/selectors/selectors.js:1348]: If you extract the function from the first comment you can also use it here :)

File: ui/components/app/permissions-connect-permission-list/permissions-connect-permission-list.js (hunk 26 above)
Comment (hmalik88) [reply ui/components/app/permissions-connect-permission-list/permissions-connect-permission-list.js:37]: Hmm yeah, I'll turn it into a closure

File: ui/components/app/snaps/snap-authorship-header/snap-authorship-header.js (hunk 27 above)
Comment (hmalik88) [reply ui/components/app/snaps/snap-authorship-header/snap-authorship-header.js:38]: Yup, will make it consistent with `snapName`

File: ui/hooks/useTransactionInsights.js (hunk 28 above)
Comment (hmalik88) [reply ui/hooks/useTransactionInsights.js:113]: Yeah, technically it should never be undefined because at that point there would be some insight snap in existence.

File: ui/selectors/selectors.js (hunk 30 above)
Comment (hmalik88) [reply ui/selectors/selectors.js:1348]: the functions from above access this selector, so not sure if that would work

File: ui/pages/confirmations/confirmation/confirmation.js (hunk 29 above)
Comment (GuillaumeRx) [reply ui/pages/confirmations/confirmation/confirmation.js:245]: Sounds good for me :)

Comment (GuillaumeRx) [reply ui/pages/confirmations/confirmation/confirmation.js:245]: Sounds good for me :)

File: ui/selectors/selectors.js (hunk 30 above)
Comment (GuillaumeRx) [reply ui/selectors/selectors.js:1348]: Yes, I realised it's not exactly equivalent

File: ui/helpers/utils/util.js [hunk 31]
```diff
@@ -31,13 +30,11 @@ import {
...
 import { Numeric } from '../../../shared/modules/Numeric';
 import { OUTDATED_BROWSER_VERSIONS } from '../constants/common';
-///: BEGIN:ONLY_INCLUDE_IF(snaps)
-import { SNAPS_METADATA } from '../../../shared/constants/snaps';
-///: END:ONLY_INCLUDE_IF
 // formatData :: ( date: <Unix Timestamp> ) -> String
 import { isEqualCaseInsensitive } from '../../../shared/modules/string-utils';
 import { hexToDecimal } from '../../../shared/modules/conversion.utils';
 import { SNAPS_VIEW_ROUTE } from '../constants/routes';
+import { stripSnapPrefix } from '@metamask/snaps-utils';
```
Comment (FrederikBolding) [inline ui/helpers/utils/util.js:37]: Why was this moved?

File: ui/components/app/permissions-connect-permission-list/permissions-connect-permission-list.js [hunk 32]
```diff
@@ -25,22 +28,21 @@ function getDescriptionNode(permission, index) {
...
+  const snapsMetadata = useSelector(getSnapsMetadata);
 
   return (
     <div className="permissions-connect-permission-list">
-      {getWeightedPermissions(t, permissions, targetSubjectMetadata).map(
-        getDescriptionNode,
-      )}
+      {getWeightedPermissions({
+        t,
+        permissions,
```
Comment (FrederikBolding) [inline ui/components/app/permissions-connect-permission-list/permissions-connect-permission-list.js:39]: Missing `subjectName`?

File: ui/components/app/permission-page-container/permission-page-container-content/permission-page-container-content.component.js [hunk 33]
```diff
@@ -6,6 +6,7 @@ import { SubjectType } from '@metamask/permission-controller';
 import PermissionsConnectHeader from '../../permissions-connect-header';
 import Tooltip from '../../../ui/tooltip';
 import PermissionsConnectPermissionList from '../../permissions-connect-permission-list';
+import { subjectMetadata } from '../../../../../.storybook/initial-states/approval-screens/token-approval';
```
Comment (FrederikBolding) [inline ui/components/app/permission-page-container/permission-page-container-content/permission-page-container-content.component.js:9]:
```suggestion
```
This looks like an auto import mistake?

**Review by FrederikBolding — APPROVED** [review]

**Review by GuillaumeRx — APPROVED** [review]
